sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala

```diff
@@ -170,6 +171,79 @@ object HiveAnalysis extends Rule[LogicalPlan] {
   }
 }
 
+/**
+ * When scanning or writing to non-partitioned Metastore Parquet tables, convert them to Parquet
+ * data source relations for better performance.
+ */
+case class ParquetConversions(sparkSession: SparkSession) extends Rule[LogicalPlan] {
```

Comment:
It might also be worth out while to create a common ancestor for these two classes.